crabs/detection_tracking/detection_utils.py

```diff
@@ -186,5 +190,127 @@ def draw_detection(
                         (0, 0, 255),
                         label_text,
                     )
-
     return image_with_boxes
+
+
+def load_ground_truth(ground_truth_paths: str) -> dict:
+    """
+    Load ground truth annotations from JSON files.
+
+    Parameters
+    ----------
+    ground_truth_paths : str
+        The directory containing the ground truth JSON files.
+
+    Returns
+    -------
+    dict
+        A dictionary mapping object IDs to their corresponding ground truth annotations.
+    """
+    ground_truths = {}
+    for filename in os.listdir(ground_truth_paths):
+        if filename.endswith(".json"):
+            object_id = filename.split("_")[-1].split(".")[
+                0
+            ]  # Extract object ID from filename
+            with open(os.path.join(ground_truth_paths, filename), "r") as f:
+                ground_truths[object_id] = json.load(f)
+    return ground_truths
+
+
+def calculate_iou(box1, box2) -> float:
+    """
+    Calculate IoU (Intersection over Union) of two bounding boxes.
+
+    Parameters:
+    box1 (list): Coordinates [x1, y1, x2, y2] of the first bounding box.
+    box2 (list): Coordinates [x1, y1, x2, y2] of the second bounding box.
+
+    Returns:
+    float: IoU value.
+    """
+    x1_box1, y1_box1, x2_box1, y2_box1 = box1
+    x1_box2, y1_box2, x2_box2, y2_box2 = box2
+
+    # Calculate intersection coordinates
+    x1_intersect = max(x1_box1, x1_box2)
+    y1_intersect = max(y1_box1, y1_box2)
+    x2_intersect = min(x2_box1, x2_box2)
+    y2_intersect = min(y2_box1, y2_box2)
+
+    # Calculate area of intersection rectangle
+    intersect_width = max(0, x2_intersect - x1_intersect + 1)
+    intersect_height = max(0, y2_intersect - y1_intersect + 1)
+    intersect_area = intersect_width * intersect_height
+
+    # Calculate area of individual bounding boxes
+    box1_area = (x2_box1 - x1_box1 + 1) * (y2_box1 - y1_box1 + 1)
+    box2_area = (x2_box2 - x1_box2 + 1) * (y2_box2 - y1_box2 + 1)
+
+    iou = intersect_area / float(box1_area + box2_area - intersect_area)
+
+    return iou
+
+
+def gt_tracking(
```

Comment:
```suggestion
def draw_gt_boxes(
```
`draw_gt_boxes` feels more explicit (otherwise with "tracking" it seems like we are running inference)